src/mapeo-project.js

```diff
@@ -749,6 +761,410 @@ export class MapeoProject extends TypedEmitter {
     return this.#iconApi
   }
 
+  /**
+   * @param {Iterable<Observation>} observations
+   * @param {Object} options
+   * @param {Set<string>} [options.seenObservations=new Set()]
+   * @param {SeenAttachments} [options.seenAttachments]
+   * @returns {AsyncIterable<Buffer | Uint8Array>}
+   */
+  async *#exportObservations(
+    observations,
+    { seenObservations = new Set(), seenAttachments = new Map() }
+  ) {
+    let first = true
+    for (const observation of observations) {
+      const { lat, lon, docId } = observation
+      if (seenObservations.has(docId)) {
+        continue
+      }
+      seenObservations.add(docId)
+      for (const attachment of observation.attachments) {
+        const { hash } = attachment
+        if (!seenAttachments.has(hash)) {
+          seenAttachments.set(hash, attachment)
+        }
+      }
+
+      let latitude = lat
+      let longitude = lon
+      let altitude = null
+      const position = observation?.metadata?.position?.coords
+      if (position) {
+        latitude = position.latitude
+        longitude = position.longitude
+        if (position.altitude !== undefined) {
+          altitude = position.altitude
+        }
+      }
```

Comment:
The lat and lon properties of an observation are the source of truth, not the position. The main situation where a different might arise is where a user has manually entered the coordinates for an observation. The position record is useful for data analysis and validation, to check any discrepancies between the location recorded by a user and the GPS data from the phone at the time. I think it's reasonable to use the position data as a fallback if lat and lon are null.